File: back/commands/commandFactory.py
from commands.connect import ConnectCommand
from commands.saveSelected import SaveSelectedCommand
from commands.requestDriver import RequestDriverCommand
from commands.hire import HireCommand
from commands.fire import FireCommand
from commands.autoContract import AutoContractCommand
from commands.swap import SwapCommand
from commands.editStats import EditStatsCommand
from commands.calendar import CalendarCommand
from commands.editContract import EditContractCommand
from commands.editPerformance import EditPerformanceCommand
from commands.editEngine import EditEngineCommand
from commands.yearSelected import YearSelectedCommand
from commands.yearSelectedH2H import YearSelectedH2HCommand
from commands.H2HConfigured import H2HConfiguredCommand
from commands.teamRequest import TeamRequestCommand
from commands.editTeam import EditTeamCommand
from commands.yearSelectedPrediction import YearSelectedPredictionCommand
from commands.yearSelectedPredictionModal import YearSelectedPredictionModalCommand
from commands.predict import PredictCommand
from commands.predictMontecarlo import PredictMontecarloCommand
from commands.unretire import UnretireCommand
from commands.configUpdate import ConfigUpdateCommand
from commands.performanceRequest import PerformanceRequestCommand
from commands.disconnect import DisconnectCommand


class CommandFactory:
    command_dict = {"connect": ConnectCommand, "saveSelected": SaveSelectedCommand, "requestDriver": RequestDriverCommand, "hire": HireCommand,
                    "fire": FireCommand, "autoContract": AutoContractCommand, "swap": SwapCommand, "editStats": EditStatsCommand,
                    "calendar": CalendarCommand, "editContract": EditContractCommand, "editPerformance": EditPerformanceCommand,
                    "editEngine": EditEngineCommand, "yearSelected": YearSelectedCommand, "yearSelectedH2H": YearSelectedH2HCommand,
                    "H2HConfigured": H2HConfiguredCommand, "teamRequest": TeamRequestCommand, "editTeam": EditTeamCommand, 
                    "yearSelectedPrediction": YearSelectedPredictionCommand, "yearSelectedPredictionModal": YearSelectedPredictionModalCommand, 
                    "predict": PredictCommand, "predictMontecarlo": PredictMontecarloCommand, "unretireDriver": UnretireCommand, 
                    "configUpdate": ConfigUpdateCommand, "performanceRequest": PerformanceRequestCommand, "disconnect": DisconnectCommand}

    def __init__(self):
        pass
        
    def create_command(self, message, client):
        command_type = message["command"]
        # print(message) #for debugging
        if command_type in self.command_dict:
            return self.command_dict[command_type](message, client)
        else:
            raise ValueError("Unknown command type")
        

File: src/js/backend/commandFactory.js
import SaveSelectedCommand from './commands/saveSelected.js';
import PerformanceRequestCommand from './commands/performanceRequest.js';
import PartRequestCommand from './commands/partRequest.js';
import DriverRequestCommand from './commands/driverRequest.js';
import EditPerformanceCommand from './commands/editPerformance.js';
import YearSelectedH2HCommand from './commands/yearSelectedH2H.js';
import ConfiguredH2HCommand from './commands/configuredH2H.js';
import CustomEnginesCommand from './commands/customEngines.js';
import TeamRequestCommand from './commands/teamRequest.js';
import YearSelectedCommand from './commands/yearSelected.js';
import EditTeamCommand from './commands/editTeam.js';
import EditStatsCommand from './commands/editStats.js';
import { editCalendar } from './scriptUtils/calendarUtils.js';
import EditCalendarCommand from './commands/editCalendar.js';

export class CommandFactory {
    constructor() {
      this.commandDict = {
        saveSelected: SaveSelectedCommand,
        driverRequest: DriverRequestCommand,
        // hire: HireCommand,
        // fire: FireCommand,
        // autoContract: AutoContractCommand,
        // swap: SwapCommand,
        editStats: EditStatsCommand,
        editCalendar: EditCalendarCommand,
        // editContract: EditContractCommand,
        editPerformance: EditPerformanceCommand,
        // editEngine: EditEngineCommand,
        yearSelected: YearSelectedCommand,
        yearSelectedH2H: YearSelectedH2HCommand,
        configuredH2H: ConfiguredH2HCommand,
        teamRequest: TeamRequestCommand,
        editTeam: EditTeamCommand,
        // unretireDriver: UnretireCommand,
        // configUpdate: ConfigUpdateCommand,
        performanceRequest: PerformanceRequestCommand,
        // fitParts: FitPartsCommand,
        partRequest: PartRequestCommand,
        customEngines: CustomEnginesCommand,
      };
    }
  
    createCommand(message) {
      const commandType = message.command;
      if (commandType in this.commandDict) {
        return new this.commandDict[commandType](message);
      } else {
        throw new Error("Unknown command type");
      }
    }
  }